### PR TITLE
change check_fill_value_and_dtype_are_compatible

### PR DESCRIPTION
**Pull request by theRealBird**
## PR Description 

`a=jax.numpy.full((2,3),-34,dtype=bool)`
The above line works fine in JAX but would have failed with Ivy. Similarly, other cases (non-exhaustively) of dtype and fill_value being castable are handled. 

Requesting a review from Ved as he seems to have worked on this assertion and my "quick fix" might have consequences.

### CODE REVIEW

**Review by vedpatwardhan — APPROVED** [review]
Good catch! Feel free to merge, thanks @theRealBird 😄

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by vedpatwardhan** [comment]
Hey @tomatillos could you please help get this PR over the line? I think the changes are done, so feel free to just resolve the conflicts and get it merged. Thanks 😄
